File: pdt-code-samples-master/paypal_pdt.pl
#!/usr/bin/perl -w
 

# It is highly recommended that you use version 6 upwards of 
# the UserAgent module since it provides for tighter server 
# certificate validation 
###
 
#
 
# PayPal PDT (Payment Data Transfer) CGI
 
#
 
###
 
 
use strict;

use CGI qw(:all unescape);
 
use CGI::Carp qw(fatalsToBrowser);
 
 
# These modules are required to make the secure HTTP request to PayPal.
 
use LWP::UserAgent 6;
 
 
 
###
 
# CUSTOMIZE THIS: This is the seller's Payment Data Transfer authorization token.
 
# Replace this with the PDT token in "Website Payment Preferences" under your account.
 
###
 
 
my $auth_token = "VUDGCF2EA5huqlEqbSLPbg0JY3F-Pokyf-99r2sZWPR4x7GkWZEa-zIG49O";
 
 
sub done_text {
 
    return (p('Your transaction has been completed, and a receipt for your purchase has been
emailed to you. You may log into your account at <a
href="https://www.paypal.com/">www.paypal.com</a> to view details of this transaction.'),
end_html());
 
}
 
 
print header(), start_html("Thank you for your purchase!");
 
 
# Set up the secure request to the PayPal server to fetch the transaction info
 
my $paypal_server = "www.paypal.com";
 
 
my $transaction = param("tx");
 
 
if (not $transaction) {
 
    print (h2("The transaction ID was not found."), done_text());
 
 
    exit();
 
}
 
 
my $paypal_url = "https://$paypal_server/cgi-bin/webscr";
 
my $query = join("&", "cmd=_notify-synch", "tx=$transaction", "at=$auth_token");
 

 
my $user_agent = new LWP::UserAgent;
 
my $request = new HTTP::Request("POST", $paypal_url);
 
 
$request->content_type("application/x-www-form-urlencoded");
$request->header(Host => $paypal_server); 
$request->content($query);
 
# Make the request
 
 
my $result = $user_agent->request($request);
 
 
if ($result->is_error) {
 
    print(h1("An error was encountered"), br(), p("An error was encountered contacting the PayPal
server:"),
 
        $result->error_as_HTML, done_text());
 
    exit();
 
}
 
 
# Decode the response into individual lines and unescape any HTML escapes
 
my @response = split("\n", unescape($result->content));
 
 
# The status is always the first line of the response.
 
my $status = shift @response;
 
 
if ($status eq "SUCCESS") {
 
    # success
 
    my %transaction;
 
 
    foreach my $response_line (@response) {
 
      my ($key, $value) = split "=", $response_line;
 
      $transaction{$key} = $value;
 
    }
 
    # These are only some of the transaction details available; there are others.
 
    # You should print all the transaction details appropriate.
 
    print(h2("Here are the details of your purchase:"),
 
      ul(li("Customer Name: " . $transaction{'first_name'} . " " . $transaction{'last_name'}),
 
          li("Item: " . $transaction{'item_name'}),
 
          li("Amount: " . $transaction{'payment_gross'})));
 
 
} elsif ($status eq "FAIL") {
 
    print(h2("Unable to retrieve transaction details."));
 
    # failure
 
} else {
 
    # unknown error
 
    print(h2("Error retrieving transaction details."));
 
}
 
 
print done_text();

File: pdt-code-samples-master/paypal_pdt.vb
// ASP VB.Net
 
Imports System.Collections.Generic
Imports System.Net
Imports System.IO 
 
Partial Public Class vbPDTSample  Inherits System.Web.UI.Page

    Protected Sub Page_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load

        ' CUSTOMIZE THIS: This is the seller's Payment Data Transfer authorization token.
        ' Replace this with the PDT token in "Website Payment Preferences" under your account.
        Dim authToken As String = "Dc7P6f0ZadXW-U1X8oxf8_vUK09EHBMD7_53IiTT-CfTpfzkN0nipFKUPYy"
        Dim txToken As String = Request.QueryString("tx")
        Dim strRequest As String = "cmd=_notify-synch&tx=" & txToken & "&at=" & authToken

        'Post back to either sandbox or live
        Dim strSandbox As String = "https://www.sandbox.paypal.com/cgi-bin/webscr"
        Dim strLive As String = "https://www.paypal.com/cgi-bin/webscr"
        Dim req As HttpWebRequest = CType(WebRequest.Create(strSandbox), HttpWebRequest)

        'Set values for the request back
        req.Method = "POST"
        req.ContentType = "application/x-www-form-urlencoded"
        req.ContentLength = strRequest.Length

        'for proxy
        'Dim proxy As New WebProxy(New System.Uri("http://url:port#"))
        'req.Proxy = proxy

        'Send the request to PayPal and get the response
        Dim streamOut As StreamWriter = New StreamWriter(req.GetRequestStream(), Encoding.ASCII)
        streamOut.Write(strRequest)
        streamOut.Close()
        Dim streamIn As StreamReader = New StreamReader(req.GetResponse().GetResponseStream())
        Dim strResponse As String = streamIn.ReadToEnd()
        streamIn.Close()

        If Not String.IsNullOrEmpty(strResponse) Then
            Dim results As New Dictionary(Of String, String)
            Dim reader As New StringReader(strResponse)
            Dim line As String = reader.ReadLine()
            If line = "SUCCESS" Then
                While True
                    Dim aLine As String
                    aLine = reader.ReadLine
                    If aLine IsNot Nothing Then
                        Dim strArr() As String
                        strArr = aLine.Split("=")
                        results.Add(strArr(0), strArr(1))
                    Else
                        Exit While
                    End If
                End While
                Response.Write("<p><h3>Your order has been received.</h3></p>")
                Response.Write("<b>Details</b><br>")
                Response.Write("<li>Name: " + results("first_name") & " " & results("last_name") & "</li>")
                Response.Write("<li>Item: " & results("item_name") & "</li>")
                Response.Write("<li>Amount: " & results("payment_gross") & "</li>")
                Response.Write("<hr>")
            ElseIf line = "FAIL" Then
                'log for manual investigation
                Response.Write("Unable to retrive transaction detail")

            End If
        Else
            Response.Write("Unknown Error")


        End If

    End Sub
End Class
